mtwitter.js

```diff
@@ -0,0 +1,244 @@
+                        cb(null);
+                    }
+                } catch (e) {
+                    console.error("Error getting bearer token:", e);
+                    cb(null);
+                }
+
+            }
+        });
+    }
+
+    this.get_bearer_token((bt) => {
+        if (bt != false) {
+            this.app_auth.bearer_token = bt;
+            console.log("Twitter Application Auth OK!");
+            this.app_twitter = new Twitter(this.app_auth);
+        } else {
+            console.error("Twitter Application Failed Auth. The bridge will be operating in a limited capacity.");
+        }
+    });
+
+    this.get_user_by_id = function(id) {
+        var ts = new Date().getTime();
+        return new Promise((resolve, reject) => {
+            for (var name in this.tuser_cache) {
+                var cached = this.tuser_cache[name];
+                if (ts - cached.cache_time > TWITTER_PROFILE_INTERVAL) {
+                    continue;
+                }
+                if (cached.user != null && cached.user.id_str == id) {
+                    resolve(cached.user);
+                }
+            }
+            var twit = this;
+            this.app_twitter.get('users/show', {
+                user_id: id
+            }, function(error, user, response) {
+                if (error) {
+                    console.error(error);
+                    reject(error);
+                } else {
+                    twit.tuser_cache[user.screen_name] = {
+                        cache_time: ts,
+                        user: user
+                    };
+                    resolve(user);
+                }
+            });
+        });
+    }
+
+    this.get_user = function(name) {
+        var twit = this;
+        console.log("Looking up @" + name);
+        return new Promise((resolve, reject) => {
+            var ts = new Date().getTime();
+            if (this.tuser_cache[name] != undefined) {
+                console.log("Checking cache for @" + name);
+                var cached = this.tuser_cache[name];
+                if (ts - cached.cache_time < TWITTER_PROFILE_INTERVAL) {
+                    resolve(cached.user);
+                }
+            }
+
+            console.log("Checking twitter for @" + name);
+            this.app_twitter.get('users/show', {
+                screen_name: name
+            }, function(error, user, response) {
+                if (error) {
+                    twit.tuser_cache[name] = {
+                        cache_time: ts,
+                        user: null
+                    };
+                    reject(error);
+                } else {
+                    twit.tuser_cache[name] = {
+                        cache_time: ts,
+                        user: user
+                    };
+                    resolve(user);
+                }
+            });
+        });
+
+    }
+
+    this.enqueue_timeline = (userid, localroom, remoteroom) => {
```

Comment:
I have to have a list of twitter users, local and remote rooms to iterate over continuously as I poll for new tweets. During startup and whenever a new timeline is requested, it gets added to the queue. I can't think of a better way to handle it.